torch/_inductor/runtime/hints.py

```diff
@@ -131,28 +135,22 @@ def create(cls, device):
             device_type = "hip"
 
         device_interface = get_interface_for_device(device)
-        if device_type in ["cuda", "hip", "xpu"]:
-            props = device_interface.get_device_properties(device)
-            return cls(
-                type=device_type,
-                index=device.index,
-                cc=device_interface.get_compute_capability(device),
-                major=props.major if hasattr(props, "major") else None,
-                regs_per_multiprocessor=props.regs_per_multiprocessor
-                if hasattr(props, "regs_per_multiprocessor")
-                else None,
-                max_threads_per_multi_processor=props.max_threads_per_multi_processor
-                if hasattr(props, "max_threads_per_multi_processor")
-                else None,
-                multi_processor_count=props.multi_processor_count
-                if hasattr(props, "multi_processor_count")
-                else None,
-                warp_size=props.warp_size if hasattr(props, "warp_size") else 32,
-            )
+        props = device_interface.get_device_properties(device)
         return cls(
             type=device_type,
             index=device.index,
             cc=device_interface.get_compute_capability(device),
+            major=getattr(props, "major", None),
+            regs_per_multiprocessor=getattr(props, "regs_per_multiprocessor", None),
+            max_threads_per_multi_processor=getattr(
+                props, "max_threads_per_multi_processor", None
+            ),
+            multi_processor_count=getattr(
+                props,
+                "multi_processor_count",
+                props.gpu_subslice_count if device_type == "xpu" else None,
+            ),
+            warp_size=getattr(props, "warp_size", 32 if device_type != "cpu" else None),
```

Comment:
@jgong5 @EikanWang which of these exist on XPU?  Is the `multi_processor_count` versus `gpu_subslice_count` thing (I copied this from ir.py) still needed?